Directory.Build.props

```diff
@@ -43,10 +43,13 @@
     <TargetFrameworkMoniker>native,Version=v0.0</TargetFrameworkMoniker>
     <!-- TODO: Keep? -->
     <!-- <RuntimeIdentifiers Condition="'$(RuntimeIdentifiers)' == ''">win;win-x64;win-x86;win-arm64</RuntimeIdentifiers> -->
+
+    <!--See https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#restore-target-->
```

Comment:
These props only apply to our repo. Can we put them somewhere users will also get them?

Reply:
Would `vnext\Directory.Build.*` work, or should this happen somewhere in `vnext\PropertySheets\External\...`?

Reply:
External property sheets are used for projects external to Microsoft.ReactNative.dll. They are included in the projects owned by the individual apps.

Imagine projects like `playground` or `e2e-test-app`.

Reply:
The templates were updated to use `PackageReference`.
I do notice that for C#, we copy the `vnext\Directory.Build.*` files into the template.
I'll do the same for the C++ project templates (those props/targets contain the VS restore logic) and merge if tests pass.